src/main/java/dev/morling/onebrc/CalculateAverage_ddimtirov.java

```diff
@@ -0,0 +1,208 @@
+/*
+ *  Copyright 2023 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.onebrc;
+
+import java.io.*;
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.ValueLayout;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+
+// gunnar morling - 2:10
+// roy van rijn -   1:01
+//                  0:37
+
+public class CalculateAverage_ddimtirov {
+    private static final String FILE = "./measurements.txt";
+
+    private static final int HASH_NO_CLASH_MODULUS = 49999;
```

Comment:
I might have been a bit too quick merging this one. How is that value determined, does it happen to prevent hash collisions just for the set of station names currently showing up in the data set? I.e. is the solution guaranteed to work also for other possible station names? If not, I'm afraid this cannot be accepted as is, as any solution must work for all names:

> Q: Can I make assumptions on the names of the weather stations showing up in the data set?
> A: No, while only a fixed set of station names is used by the data set generator, any solution should work with arbitrary UTF-8 station names (for the sake of simplicity, names are guaranteed to contain no ; character).

Reply:
This may work for this dataset, but not for other large datasets where hash collisions are common.